docs/PLUGIN_AUTHORING.md

```diff
@@ -17,152 +17,134 @@ See the License for the specific language governing permissions and
 
-Like any good recipe, it will do you well to read the entire document, as well as the [Plugin Best Practices](https://github.com/intelsdi-x/snap/blob/master/docs/PLUGIN_BEST_PRACTICES.md), before you start cooking.
+1. [Overview](#overview)
+2. [Developing Plugins](#developing-plugins)
+    * [Plugin Type](#plugin-type)
+    * [Plugin Name](#plugin-name)
+    * [Plugin Metric Namespace](#plugin-metric-namespace)
+    * [Plugin Interface](#plugin-interface)
+    * [Plugin Version](#plugin-version)
+    * [Plugin Release](#plugin-release)
+    * [Plugin Metadata](#plugin-metadata)
+    * [Documentation](#documentation)
 
-Bon Appétit! :stew:
+## Overview
 
-## Plugin Authoring
-Snap itself runs as a master daemon with the core functionality that may load and unload plugin processes via either CLI or HTTP APIs.
+Snap daemon runs as a service which provides the core functionality such as plugin management and task scheduling. Snap plugins extend Snap by providing additional capability to gather, transform, or publish metrics. Since plugins are standalone programs, they can be written in any language.
 
-A Snap plugin is a program, or a set of functions or services, written in Go or any language; that may seamlessly integrate with snap as executables. 
+The following libraries are available to simplify the process of writing a plugin:
 
-Communication between Snap and plugins uses RPC either through HTTP or TCP protocols. HTTP gRPC is good for any language to use that gRPC supports (see [gRPC docs](http://grpc.io/docs)) while the native client is only suitable for plugins written in Golang. When a plugin is written using one of the available snap-plugin-libs ([snap-plugin-lib-go](https://github.com/intelsdi-x/snap-plugin-lib-go), [snap-plugin-lib-py](https://github.com/intelsdi-x/snap-plugin-lib-py), or [snap-plugin-lib-cpp](https://github.com/intelsdi-x/snap-plugin-lib-cpp)) Google Protobuf, a binary serialization format, is used to encode/decode the data (see [plugin.proto](https://github.com/intelsdi-x/snap/blob/master/control/plugin/rpc/plugin.proto)). 
+* [snap-plugin-lib-go](https://github.com/intelsdi-x/snap-plugin-lib-go)
+* [snap-plugin-lib-cpp](https://github.com/intelsdi-x/snap-plugin-lib-cpp)
+* [snap-plugin-lib-py](https://github.com/intelsdi-x/snap-plugin-lib-py)
 
-Before starting writing Snap plugins, check out the [Plugin Catalog](https://github.com/intelsdi-x/snap/blob/master/docs/PLUGIN_CATALOG.md) to see if any suit your needs. If not, you need to reference the plugin packages that defines the type of structures and interfaces inside snap and then write plugin endpoints to implement the defined interfaces.
+Communication between Snap daemon and plugins uses gRPC. So even if a plugin library isn't available in the language of your choice, you can still write a plugin using [gRPC library](http://grpc.io/docs) as a starting point. However this requires additional knowledge about Snap API, and is beyond the scope of this document.
```

Comment:
Should we have an advanced section? Anything you recommend preserving and copying over?